lsp-types/src/Language/LSP/Protocol/Message/Types.hs

```diff
@@ -132,7 +132,7 @@ data TResponseMessage (m :: Method f Request) =
   TResponseMessage
     { _jsonrpc :: Text
     , _id      :: Maybe (LspId m)
-    , _result  :: Either (TResponseError m) (MessageResult m)
+    , _result  :: Either ResponseError (MessageResult m)
```

Comment:
```suggestion
    -- TODO: use `TResponseError m` for the error type, this will require quite a lot of adaptation downstream
    , _result  :: Either ResponseError (MessageResult m)
```